server/src/main/java/org/opensearch/action/admin/cluster/settings/TransportClusterUpdateSettingsAction.java

```diff
@@ -270,6 +274,13 @@ public ClusterState execute(final ClusterState currentState) {
                         logger
                     );
                     changed = clusterState != currentState;
+                    // Remote Store migration: Checks if the applied cluster settings
+                    // has switched the cluster to STRICT mode. If so, checks and applies
+                    // appropriate index settings depending on the current set of node types
+                    // in the cluster
+                    if (isSwitchToStrictCompatibilityMode(clusterState.metadata().settings())) {
+                        return finalizeMigration(clusterState);
+                    }
```

Comment:
We need to update `changed` as well , as it does a reroute on basis of that .

Reply:
We are just updating `IndexMetadata` here and nothing on the `RoutingTable` or `RoutingNodes`. The `changed` flag is already set just before this invocation since without changes in `RoutingTable` or `RoutingNodes`, the `reroute` invocation would be a no-op

Reply:
Addressed in new commit